test/functional.js

```diff
@@ -500,27 +500,48 @@ describe('Functional tests using webpack', function() {
             config.enableVersioning(true);
 
             testSetup.runWebpack(config, (webpackAssert) => {
-                if (!process.env.DISABLE_UNSTABLE_CHECKS) {
-                    expect(config.outputPath).to.be.a.directory()
-                        .with.files([
-                            '0.590a68c7.js', // chunks are also versioned
-                            '0.8ec31654.css',
-                            'main.4a5effdb.js',
-                            'h1.8ec31654.css',
-                            'bg.0ec2735b.css',
-                            'manifest.json',
-                            'entrypoints.json',
-                            'runtime.a59dcd0a.js',
-                        ]);
-                }
+                const actualFilesMap = {};
+                const expectedFiles = [
+                    '0.[hash].js', // chunks are also versioned
+                    '0.[hash].css',
+                    'main.[hash].js',
+                    'h1.[hash].css',
+                    'bg.[hash].css',
+                    'manifest.json',
+                    'entrypoints.json',
+                    'runtime.[hash].js',
+                ];
+
+                expect(config.outputPath).to.be.a.directory()
+                    .with.files.that.satisfy(files => {
+                        for (const expectedFile of expectedFiles) {
+                            const expectedFileRegexp = new RegExp(`^${
+                                expectedFile
+                                    .replace('.', '\\.')
+                                    .replace('[hash]', '[0-9a-f]+')
+                            }$`);
+
+                            const matchingFile = files.find(file => {
+                                return expectedFileRegexp.test(file);
+                            });
+
+                            if (!matchingFile) {
+                                return false;
+                            }
+
+                            actualFilesMap[expectedFile] = matchingFile;
+                        }
+
+                        return true;
+                    });
```

Comment:
So the unstable hashes only affect this test? Is it the only one where we have versioning enabled and are checking for CSS paths? If so, I'm ok cool with this... it's not ideal... because we'd like to know if hashes are changing unnecessarily... but we need to be pragmatic and nobody can think of a better solution :). 

Could we refactor this into a reusable function... more for clarity than reusability - something on `webpackAssert`?